test/extensions/filters/http/common/fuzz/uber_filter.h

```diff
@@ -85,14 +85,29 @@ class UberFilterFuzzer {
       return;
     }
 
-    for (int i = 0; i < data.data().size(); i++) {
-      if (i == data.data().size() - 1 && !data.has_trailers()) {
-        end_stream = true;
+    if (data.has_http_body()) {
+      for (int i = 0; i < data.http_body().data_size(); i++) {
+        if (i == data.http_body().data_size() - 1 && !data.has_trailers()) {
+          end_stream = true;
+        }
+        Buffer::OwnedImpl buffer(data.http_body().data(i));
+        ENVOY_LOG_MISC(debug, "Decoding http data: {} ", buffer.toString());
+        if (filter->decodeData(buffer, end_stream) != Http::FilterDataStatus::Continue) {
+          return;
+        }
       }
-      Buffer::OwnedImpl buffer(data.data().Get(i));
-      ENVOY_LOG_MISC(debug, "Decoding data: {} ", buffer.toString());
-      if (filter->decodeData(buffer, end_stream) != Http::FilterDataStatus::Continue) {
-        return;
+    } else if (data.has_proto_body()) {
+      const std::string serialized = data.proto_body().message().SerializeAsString();
+      for (size_t i = 0; i < serialized.length(); i += data.proto_body().chuck_size()) {
+        const std::string chunk = serialized.substr(i, data.proto_body().chuck_size());
+        if (i + data.proto_body().chuck_size() >= serialized.length() && !data.has_trailers()) {
```

Comment:
Question for reviewers: Is there a cleaner way to do this?

Reply:
for loop over all but the last of
`absl::StrSplit(serialized, ByLength()data.proto_body().chuck_size())`?, handle the last separately?

Reply:
Thanks, will do